RealmSwift/Logger.swift

```diff
@@ -0,0 +1,275 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2024 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import Realm
+import Realm.Private
+
+/**
+ `Logger` is used for creating your own custom logging logic.
+
+ You can define your own logger creating an instance of `Logger` and define the log function which will be
+ invoked whenever there is a log message.
+
+ ```swift
+ let logger = Logger(level: .all, category: Category.realm) { level, message in
+    print("Realm Log - \(category.toString())-\(level): \(message)")
+ }
+ ```
+
+ Set this custom logger as you default logger using `Logger.shared`.
+
+ ```swift
+    Logger.shared = inMemoryLogger
+ ```
+
+ - note: By default default log threshold level is `.info`, and logging strings are output to Apple System Logger.
+*/
+public typealias Logger = RLMLogger
+extension Logger {
+    /**
+     Log a message to the supplied level.
+
+     ```swift
+     let logger = Logger(level: .info, logFunction: { level, message in
+         print("Realm Log - \(level): \(message)")
+     })
+     logger.log(level: .info, message: "Info DB: Database opened succesfully")
+     ```
+
+     - parameter level: The log level for the message.
+     - parameter category: The log category for the message.
+     - parameter message: The message to log.
+     */
+    internal func log(level: LogLevel, category: LogCategory = Category.realm, message: String) {
```

Comment:
Why is this here?

Reply:
This just makes it easier for us internally to log  in the SDK side, we do have a category for this kind of logs Category.sdk in Swift

Reply:
There is also a category for app logs Category.app, so we could expose this to the user in that matters, but I agree that we don't need to expose this to the users and add noise to our logging mechanism, but I do agree that we could have some logging in the sdk side, we should discuss how granular we want this because we don't want to contaminate Core's log